### DISCUSSION TITLE
Add a chord mode that stays in key

### DISCUSSION THREAD

**Discussion by panicmechanic** [SynthstromAudible/DelugeFirmware#1766] [Ideas]
### Please describe the problem:

When recording using (any) scale and chord mode, the firmware appears to create and switch to a "user-defined" scale at 1 bar, then modifies it again at the second bar. 

After this second modification, it's a full chromatic scale. No more changes are made.

If you have chord mode set, but unused, and record single notes as normal - the bug won't occur. But as soon as you begin using one of the chord types, the bug appears.

I've recreated the problem on the last 3 nightly builds. 

### What is the expected behavior?

For the scale to remain unchanged throughout recording. 

### What hardware did you reproduce it with?

OLED

### What firmware did you reproduce it with?

Nightly

### What is the firmware name:

deluge-v1_2_0-nightly+2024_04_22-ba9747e

### If possible provide the steps to reproduce the issue and upload additional media:

**30-sec demonstration of bug here:** https://i.imgur.com/ZMysONQ.mp4

**Steps:**

1. Turn on deluge
2. Use any synth
3. Open isomorphic keyboard
4. Switch to in-key mode (it occurs in both, but is more visually obvious in this mode)
5. Set to chord mode, pick any chord
6. Hit record, play
7. Press any chords
8. Continue until you have a full chromatic user-defined scale

**Comment by m-m-adams** [comment]
This occurs because the chord mode will play out of scale notes - e.g. a C minor chord over a C major scale will have an Eb, and then the Eb will be added to the scale since Deluge doesn't handle accidentals. For a current workaround you can disable scale mode on the clip - this prevents notes in this clip from affecting the scale of the deluge.


Longer term we probably want a few different things 
1. Better accidental support, so playing out of scale notes doesn't ruin the 8 note scale
2. A chord mode based on scale degrees (like chord sheets) rather than explicit chord choices
3. Potentially a "best fit" chord where we shift a note to make a different chord that's in scale? It would need some thought on whether C minor would become C major or C sus2 etc.
